addons/crm/models/crm_lead.py

```diff
@@ -930,15 +931,91 @@ def action_schedule_meeting(self):
         partner_ids = self.env.user.partner_id.ids
         if self.partner_id:
             partner_ids.append(self.partner_id.id)
+        current_opportunity_id = self.id if self.type == 'opportunity' else False
         action['context'] = {
-            'default_opportunity_id': self.id if self.type == 'opportunity' else False,
+            'default_opportunity_id': current_opportunity_id,
             'default_partner_id': self.partner_id.id,
             'default_partner_ids': partner_ids,
             'default_team_id': self.team_id.id,
             'default_name': self.name,
         }
+
+        # 'Smart' calendar view : get the most relevant time period to display to the user.
+        if current_opportunity_id:
+            smart_mode, smart_initial_date = self._get_opportunity_meeting_view_parameters(current_opportunity_id)
+            action['context'].update({'default_mode': smart_mode, 'initial_date': smart_initial_date})
```

Comment:
I would rename it display_mode and initial_date, you already explain that's for smart calendar.